pkg/util/deployment/deployment.go

```diff
@@ -236,6 +236,10 @@ func addHashKeyToRSAndPods(deployment *extensions.Deployment, c clientset.Interf
 		return updatedRS, nil
 	}
 
+	if err = waitForPodsHashPopulated(c, namespace, updatedRS.Name); err != nil {
```

Comment:
You are right. Because we waited replicaset controller to update the observed version. I'll add the comment.